sdk/keyvault/keyvault-admin/src/lro/selectiveRestore/operation.ts

```diff
@@ -209,14 +209,19 @@ async function update(
   }
 
   if (!state.isCompleted) {
-    const selectiveRestoreOperation = await fullRestoreStatus(client, vaultUrl, state.jobId, {
+    const selectiveRestoreOperation = await restoreStatus(client, vaultUrl, state.jobId, {
       requestOptions
     });
-    const { endTime, error } = selectiveRestoreOperation;
+    const { endTime, status, statusDetails, error } = selectiveRestoreOperation;
+
+    state.endTime = endTime;
+    state.status = status;
+    state.statusDetails = statusDetails;
```

Comment:
Note that this PR shouldn't have been about source code changes, but it ends up involving them due to bugs found as I tested.

Also note that, even though this whole file is considerably similar to the restore operation file (noting that this is the selectiveRestore operation file), I don't want to preemptively unify them. Unification here would suggest me making some form of "common" set of resources for LRO operations, which is something I'm not sure we should do at the moment. We might go that route in the future for sure.